internal/provider/utils.go

```diff
@@ -41,14 +42,19 @@ func CheckDn(ctx context.Context, client *client.Client, dn string, diags *diag.
 	}
 }
 
-func DoRestRequest(ctx context.Context, diags *diag.Diagnostics, client *client.Client, path, method string, payload *container.Container) *container.Container {
+func DoRestRequest(ctx context.Context, diags *diag.Diagnostics, client *client.Client, path, method string, payload *container.Container, escapeHtml bool) *container.Container {
 
 	// Ensure path starts with a slash to assure signature is created correctly
 	if !strings.HasPrefix("/", path) {
 		path = fmt.Sprintf("/%s", path)
 	}
-
-	restRequest, err := client.MakeRestRequest(method, path, payload, true)
+	var restRequest *http.Request
+	var err error
+	if escapeHtml {
+		restRequest, err = client.MakeRestRequest(method, path, payload, true)
+	} else {
+		restRequest, err = client.MakeRestRequestRaw(method, path, payload.EncodeJSON(), true)
```

Comment:
How does this EncodeJson() function exactly work, because when I look at the function it mentions it does not escape by default but it can take a range of encoder options, like `EncodeOptHTMLEscape`. We never seem to pass in any options, would this function always set escaping to false?

Reply:
EncodeJson() avoids calling the string() function which internally calls unmarshall() (it again converts unique characters to unicode)